native/src/modules/layout/components/Header.js

```diff
@@ -70,35 +71,47 @@ type PropsType = {|
   goToLanguageChange?: () => void,
   routeCityModel?: CityModel,
   language: string,
+  path: string | null,
+  routeMappingType: string | null,
   dispatch: Dispatch<StoreActionType>
 |}
 
-class Header extends React.PureComponent<PropsType> {
-  canGoBackInStack (): boolean {
-    return !!this.props.previous
-  }
+const Header = (props: PropsType) => {
+  const canGoBackInStack = useCallback(() => {
+    return !!props.previous
+  })
 
-  goBackInStack = () => {
-    this.props.navigation.goBack()
-  }
+  const goBackInStack = useCallback(() => {
+    props.navigation.goBack()
+  })
 
-  goToLanding = () => {
-    const { navigation, dispatch } = this.props
+  const goToLanding = useCallback(() => {
+    const { navigation, dispatch } = props
     // $FlowFixMe Navigation type of the header does not match that of screens.
     createNavigateToLanding(dispatch, navigation)()
-  }
+  })
 
-  goToSettings = () => {
-    this.props.navigation.navigate(SETTINGS_ROUTE)
-  }
+  const goToSettings = useCallback(() => {
+    props.navigation.navigate(SETTINGS_ROUTE)
+  })
 
-  onShare = async () => {
-    const { scene, t } = this.props
-    const shareUrl = scene.route.params?.shareUrl
-    if (!shareUrl) { // The share option should only be shown if there is a shareUrl
+  const onShare = useCallback(async () => {
+    const { scene, t, path, language, routeMappingType } = props
+    const shareUrlFromScene = scene.route.params?.shareUrl
```

Comment:
~~I think we should remove any trace of sharing from the navigation props and only use the state.
I think we can just get the current route_key and then use it to search the state. If we do not find the proper information then do not enable the sharing.~~

Reply:
Oh oke we have some route which are not in the state correct...
I think we als have there a way of changing the language right? Does this actually work like this? Or do we have a problem with not-updating share urls in non-state routes too?

Reply:
Yeah it is not working, too.

Reply:
Routes without a mapping in the redux state are working fine already as it is not possible to change languages there.